### PR TITLE
NWN2: Implement a FeatAdd script call and check feat requirements

### PR DESCRIPTION
**Pull request by rjshae**
This implements the FeatAdd scripting language function, adds a featAdd
call to the Creature class, and checks that the feat requirement have been
satisfied by the creature.

### CODE REVIEW

File: src/engines/nwn2/feats.cpp [hunk 1]
```diff
@@ -132,6 +136,55 @@ bool Feats::getHasCustomFeat(Custom feat) const {
 	return _hasCustomFeat[feat];
 }
 
+/** Return true only if the creature satisfies the feat requirements */
+bool Feats::meetsRequirements(Creature *creature, uint32 id) const {
+	static const Common::UString kFeatMinCols[] = {"MINSTR", "MINDEX", "MINCON", "MININT", "MINWIS", "MINCHA"};
+	static const Common::UString kFeatMaxCols[] = {"MAXSTR", "MAXDEX", "MAXCON", "MAXINT", "MAXWIS", "MAXCHA"};
+	static const Common::UString kFeatPrereq1 = "PREREQFEAT1", kFeatPrereq2 = "PREREQFEAT2";
+	uint i;
```
Comment (DrMcCoy) [inline src/engines/nwn2/feats.cpp]: Move that loop variable into the for-loop, please

Comment (rjshae) [reply src/engines/nwn2/feats.cpp]: I've attempted to address your concern. Thanks.

File: src/engines/nwn2/script/functions.h [hunk 2]
```diff
@@ -246,6 +246,8 @@ class Functions {
 	void getSkillRank   (Aurora::NWScript::FunctionContext &ctx);
 	void getHasFeat     (Aurora::NWScript::FunctionContext &ctx);
 
+	void featAdd        (Aurora::NWScript::FunctionContext &ctx);
```
Comment (DrMcCoy) [inline src/engines/nwn2/script/functions.h]: If you're separating it by a blank link, you can drop that extra alignment in front of the opening paren there. It's supposed to line up each sub-group.

**Comment by rjshae** [comment]
Code updated per review.

**Comment by DrMcCoy** [comment]
Merged as e1dc0d29a...5bbe8d9a6, thanks! :)

**Pull request closed without merging**